### PR TITLE
ansible-config: add simple tests

### PR DESCRIPTION
**Pull request by pilou-**
##### SUMMARY
ansible-config: add simple tests

Related: #34673

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
ansible-config

##### ANSIBLE VERSION
```
2.5
```

### CODE REVIEW

**Comment by pilou-** [comment]
cc @maxamillion

**Review by gundalow — APPROVED** [review]
Nice. Thanks for adding this

**Comment by mattclay** [comment]
CI failure in the `ansible` integration test:

```
2018-01-15 22:21:32 + set -o pipefail
2018-01-15 22:21:32 + ansible-config -c ./ansible-test.cfg view
2018-01-15 22:21:32 + grep 'remote_user = admin'
2018-01-15 22:21:34 remote_user = admin
2018-01-15 22:21:34 + ansible-config -c ./ansible-test.cfg dump
2018-01-15 22:21:34 + grep '^DEFAULT_REMOTE_USER([^)]*) = admin$'
2018-01-15 22:21:37 Command exited with status 1 after 13.6531560421 seconds.
```

File: test/integration/targets/ansible/runme.sh [hunk 1]
```diff
@@ -7,3 +7,12 @@ ansible --help
 
 ansible testhost -i ../../inventory -m ping  "$@"
 ansible testhost -i ../../inventory -m setup "$@"
+
+set -o pipefail
```
Comment (mattclay) [inline test/integration/targets/ansible/runme.sh]: Go ahead and set this at the start of the script.

File: test/integration/targets/ansible/runme.sh [hunk 2]
```diff
@@ -7,3 +7,12 @@ ansible --help
...
+ansible-config -c ./ansible-test.cfg view | grep 'remote_user = admin'
+ansible-config -c ./ansible-test.cfg dump | grep '^DEFAULT_REMOTE_USER([^)]*) = admin$'
+ansible-config list | grep 'DEFAULT_REMOTE_USER'
+set +o pipefail
```
Comment (mattclay) [inline test/integration/targets/ansible/runme.sh]: Might as well leave pipefail on for the entire script.

**Review by mattclay — CHANGES_REQUESTED** [review]

File: lib/ansible/config/manager.py [hunk 3]
```diff
@@ -116,7 +116,7 @@ def get_config_type(cfile):
 
     ftype = None
     if cfile is not None:
-        ext = to_text(os.path.splitext(cfile)[-1])
+        ext = os.path.splitext(cfile)[-1]
```
Comment (maxamillion) [inline lib/ansible/config/manager.py:119]: What's the motivation for removing `to_text` here?

File: test/integration/targets/ansible/runme.sh (hunk 1 above)
Comment (maxamillion) [reply test/integration/targets/ansible/runme.sh]: +1 to @mattclay

File: lib/ansible/config/manager.py (hunk 3 above)
Comment (pilou-) [reply lib/ansible/config/manager.py:119]: The first commit of this pull-request reverts a2596f888700c7d97201c269ce6348830b46d6de.

**Comment by pilou-** [comment]
ready_for_review

**Review by mattclay — APPROVED** [review]

File: lib/ansible/config/manager.py (hunk 3 above)
Comment (maxamillion) [reply lib/ansible/config/manager.py:119]: Yes, but why?

Comment (pilou-) [reply lib/ansible/config/manager.py:119]: Using Python 3 and a2596f8, `ansible-config -c ansible.cfg dump` remains unusable. a2596f8 fixes an exception but the configuration file is still not read: [`ConfigParser.read`](https://github.com/python/cpython/blob/master/Lib/configparser.py#L690) doesn't support bytes:
```
(Pdb) bt
  ansible/bin/ansible-config(118)<module>()
-> exit_code = cli.run()
  ansible/lib/ansible/cli/config.py(72)run()
-> self.config = ConfigManager(self.config_file)
  ansible/lib/ansible/config/manager.py(202)__init__()
-> self._parse_config_file()
  ansible/lib/ansible/config/manager.py(220)_parse_config_file()
-> self._parser.read(cfile)
> /usr/lib/python3.6/configparser.py(696)read()
-> with open(filename, encoding=encoding) as fp:
```
```
(Pdb) filenames
b'path/to/ansible.cfg'
(Pdb) [x for x in filenames]
[... 47, 97, 110, 115, 105, 98, 108, 101, 46, 99, 102, 103]
```
Avoiding `bytes` conversion fixes both problem, then the conversion added by a2596f8 becomes useless.

Comment (maxamillion) [reply lib/ansible/config/manager.py:119]: Alright, the "avoiding `bytes` conversion fixes both problems" is the part I didn't realize. :+1:

**Comment by mscherer** [comment]
shipit

**Pull request merged**